crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -451,9 +482,17 @@ impl CheckpointExecutor {
             .map(|digests| (digests.transaction, digests.effects))
             .unzip();
 
-        self.epoch_store
-            .insert_finalized_transactions(&tx_digests, sequence_number)
-            .expect("failed to insert finalized transactions");
+        pipeline_handle
+            .skip_to(PipelineStage::FinalizeTransactions)
+            .await;
+
+        // Currently this code only runs on validators, where this method call does nothing.
+        // But in the future, fullnodes may follow the mysticeti dag and build their own checkpoints.
+        self.insert_finalized_transactions(&tx_digests, sequence_number);
+
+        pipeline_handle
+            .skip_to(PipelineStage::ProcessCheckpointData)
+            .await;
 
```

Comment:
I must be missing it but what finishes the `ProcessCheckpointData` stage in this path, as happens in line 565 on the other path?

Reply:
i broke this while getting it ready for review, fixing